### DISCUSSION TITLE
Is HCL the same as LCH and if so is there a way to get similar numbers to Gimp and online converters

### DISCUSSION THREAD

**Discussion by jarj1** [ImageMagick/ImageMagick#7118] [Help] [answered]
### ImageMagick version

7.0.10-18

### Operating system, version and so on

Win 10 Q16 x64

### Description

While working with some images in gimp and with Imagemagick I've noticed IM's HCL and Gimps LCH can be quite different.  I have checked an online tool and it seems to match with Gimp - are they both wrong and IM correct? Or have I done something wrong

Example below
`magick -size 1x1 canvas:"rgb(163,217,167)" -colorspace HCL  txt:-`
XX ImageMagick pixel enumeration: 1,1,65535,hcl
0,0: (124.444,21.1765%,76.5275%)  #587E3636C3E8  hcl(124.444,21.1765%,76.5275%)

IM:
    L : 76.5275%
    C : 21.1765%
    H : 124.444

Gimp color picker
![163_217_167](https://github.com/ImageMagick/ImageMagick/assets/66709392/acb71f12-e8c9-472a-9fcc-748e56476b77)

Using an online converter from [http://colormine.org/convert/rgb-to-lch](http://colormine.org/convert/rgb-to-lch)
![163_217_167_cm](https://github.com/ImageMagick/ImageMagick/assets/66709392/6743fb13-becd-4b81-b563-573bd78f7372)

**Comment by snibgo** [comment]
HCL and LCH are different colorspaces, so it would seem better to compare IM's LCH with Gimp's LCH. They are different. I don't know why.

**Reply by jarj1** [reply]
Thanks snibgo, 
my mistake for thinking HCL and LCH were the same

I have re-run the script with LCH but I can't make sense of the numbers that came out (Is LCH = CIELCH?) :
`magick  -size 1x1 canvas:"rgb(163,217,167)" -colorspace LCH   txt:-`
X ImageMagick pixel enumeration: 1,1,65535,lch
0,0: (53704.1,41293,26358.1)  #D1C8A14D66F6  lch(208.965,160.673,102.561)

It is a bit long winded but I have found a way around to get L* and C* using CIELab with the fact that C*=(a^2 + b^2)^0.5
`magick  -size 1x1 canvas:"rgb(163,217,167)" -colorspace CIELab   txt:-`
X ImageMagick pixel enumeration: 1,1,65535,cielab
0,0: (53704.1,25801.6,4915.38)  #D1C864CA9333  cielab(81.9471,-27.1045,19.126)

L* = 81.9471
C* = 33.17314872  `pow(pow(-27.1045,2) + pow(19.126,2), 0.5) `

**Comment by urban-warrior** [comment, marked as answer]
Thank you for reporting the issue. We have successfully reproduced it and are actively working on a patch to resolve it. You can expect this patch to be merged into the [main GIT branch](https://github.com/ImageMagick/ImageMagick), later today. As part of our commitment to quality, this fix will also be included in the upcoming [beta releases of ImageMagick](https://imagemagick.org/archive/beta/) by tomorrow. Your patience and feedback are greatly appreciated.

Note that while the LCH color space transform was functioning correctly, the values were being transformed to fit within the range of [0...65535]. As a result, the TXT coder was outputting the shifted LCH pixel values. We have since rectified this issue and ensured that the output now aligns with the expected values.
